### PR TITLE
feat: drawer handle component

### PR DESCRIPTION
**Pull request by rortan134**
An optional complementary component. Implements [Apple's grabber practices](https://developer.apple.com/design/human-interface-guidelines/sheets#Platform-considerations):

`<Drawer.Handle />`

- Tapping on the handle cycles through the snap points if present, if not it closes it.
- Double tapping on the handle immediately closes the drawer.

I added some style defaults but let me know if it's better without

### CODE REVIEW

File: src/use-snap-points.ts [hunk 1]
```diff
@@ -102,13 +102,15 @@ export function useSnapPoints({
...
+    if (activeSnapPoint || activeSnapPointProp) {
+      const newIndex =
+        snapPoints?.findIndex((snapPoint) => snapPoint === activeSnapPointProp || snapPoint === activeSnapPoint) ??
+        null;
       if (snapPointsOffset && newIndex && typeof snapPointsOffset[newIndex] === 'number') {
         snapToPoint(snapPointsOffset[newIndex] as number);
       }
     }
-  }, [activeSnapPointProp, snapPoints, snapPointsOffset, snapToPoint]);
+  }, [activeSnapPoint, activeSnapPointProp, snapPoints, snapPointsOffset, snapToPoint]);
```
Comment (rortan134) [inline src/use-snap-points.ts:113]: This is needed in order to `setActiveSnapPoint` updates to sync

File: src/index.tsx [hunk 2]
```diff
@@ -672,6 +673,42 @@ function Root({
...
+      closeDrawer();
+      return;
+    }
+
+    const nextSnapPoint = snapPoints[snapPoints.findIndex((point) => point === activeSnapPoint) + 1];
+    setActiveSnapPoint(nextSnapPoint);
+  }
+
+  return (
+    <span
```
Comment (rortan134) [inline src/index.tsx:699]: This might be better as a button tbh

**Comment by mlshv** [comment]
After attempting to integrate the drawer with my scrollable content, I've realized there are issues I'd prefer not to handle. The handle component addresses these by clearly separating content scrolling from drawer drag gestures.

Can you update this handle component to allow the drawer to be dragged exclusively by the handle?

**Comment by rortan134** [comment]
> Can you update this handle component to allow the drawer to be dragged exclusively by the handle?

I think that's alright as long as @emilkowalski is fine with it

**Comment by mlshv** [comment]
@emilkowalski 

image

**Comment by emilkowalski** [comment]
@rortan134 This is so cool, could you update it based on @mlshv suggestions? We can also update the readme

**Comment by mlshv** [comment]
@rortan134 hey, do you need help with this? I am very interested in this being shipped, would be glad to assist 😌

**Comment by rortan134** [comment]
I think a more composable approach would be to have a drawer.header that would handle the gestures and let the handle have its own interactions (tapping, etc.). This is fine though

chrome_shflXiBmiA chrome_gowIFhAq2I

@emilkowalski

**Comment by devjmetivier** [comment]
@rortan134 Great PR! I was just browsing the repo to see if I could contribute this.

I love your thoughts about on the more composable approach. If I could make one suggestion in referring to it as a "handle" instead of a "header". In the same way that `Drawer.Trigger` describes the utility of the trigger, I believe `Drawer.Handle` would do the same.

Let me know if I can help getting this across the finish line!

**Comment by emilkowalski** [comment]
@rortan134 I don't fully get your commen,t but the video on the right where the whole header is draggable makes more sense to me. Is this PR ready?

**Comment by rortan134** [comment]
Should be, although the first vid is the default

**Comment by emilkowalski** [comment]
Hmm... seems like some tests are failing. @rortan134 Could you take a look please?

**Comment by JE-lee** [comment]
It looks good. I hope to have this feature soon.

**Comment by emilkowalski** [comment]
Can we fix merge conflicts? Looks good otherwise!

**Comment by fabiodinota** [comment]
Don't wanna rush but would really love this feature so if I can do anything to help get this shipped, please let me know.

**Comment by arunavo4** [comment]
I would like to use this drawer below the content, can't seem to get it to work since my drawer direction is top.

**Pull request merged**